compiler/rustc_smir/src/rustc_internal/internal.rs

```diff
@@ -166,8 +166,10 @@ impl RustcInternal for FloatTy {
 
     fn internal<'tcx>(&self, _tables: &mut Tables<'_>, _tcx: TyCtxt<'tcx>) -> Self::T<'tcx> {
         match self {
+            FloatTy::F16 => rustc_ty::FloatTy::F16,
```

Comment:
I'd rather leave f16/f128 out of stable mir altogether, since it's not stable at all. Just make it ICE if you try to convert it.

Reply:
Sure, set these to `unimplemented!`